### ISSUE TITLE
Return type cannot be 'void'

### ISSUE THREAD

**Issue by vukisz** [milnet92/MXTXppInterpreter#47]
Very strange issue on new custom CustTable extension methods. If I try to print them or use I am getting "Return type cannot be 'void'" error. Other STD methods works allright.

More context in screenshot (with obfuscated object prefixes)
![image](https://github.com/user-attachments/assets/851850e5-2b94-4331-b5cb-af4edc024edd)

**Comment by milnet92** [comment]
Hi Vulkz, I'll try to replicate the same scenario. Could you show me the EDT metadata of your return type (*FCBalance). I'll try to replicate the issue.

**Comment by vukisz** [comment]
> Hi Vulkz, I'll try to replicate the same scenario. Could you show me the EDT metadata of your return type (*FCBalance).

Sure (nothing special there and is not working with any extension method and other types):
![image](https://github.com/user-attachments/assets/eee5f3f1-c39b-47b6-a647-5625116ee48f)

**Comment by milnet92** [comment]
So I tested your exact scenario and seems related to some kind of cache. The error you are seeing is due to SysDictTable class not returning your newly created method. *Return type cannot be 'void'* under the hood means that as an expression, is expected to be return something and since the method "does not exists" (due to the cache issue), no type is infered. So, **this only affects the extension methods created recently on a Dev environment**.

I did a test myself (on a Unified Dev environment) and happened to me one time, seems to be kinda "random" but after some time I could call the method without any problem... maybe It's worth exploring the use of MetaModel reflection methods instead of the SysDict framework and see if we can avoid this error.

If you pull the latests changes from yesterday you will se another error like *'Type 'X' does not have a method called 'Y'* which is a bit more coherent of what is happening.
